pkg/frontend/querymiddleware/codec_test.go

```diff
@@ -677,7 +677,7 @@ func mockPrometheusResponse(numSeries, numSamplesPerSeries int) *PrometheusRespo
 	return &PrometheusResponse{
 		Status: "success",
 		Data: &PrometheusData{
-			ResultType: "vector",
+			ResultType: "matrix",
```

Comment:
I'm trying to understand where this change comes from? That's completely unrelated to the marshaling function, right?

Reply:
This previous `vector` result type value prevented `BenchmarkPrometheusCodec_DecodeResponse` from being run. 

The following error was prompted:

```
BenchmarkPrometheusCodec_DecodeResponse
    codec_test.go:618: 
        	Error Trace:	/Users/ortuman/Projects/mimir/pkg/frontend/querymiddleware/codec_test.go:618
        	Error:      	Received unexpected error:
        	            	querymiddleware.PrometheusResponse.Data: can't marshal prometheus result type "value"
        	Test:       	BenchmarkPrometheusCodec_DecodeResponse
--- FAIL: BenchmarkPrometheusCodec_DecodeResponse
FAIL
```

Reply:
thanks for explaining